### PR TITLE
Feature/update list

### LINKED ISSUE #304: List component review

### PR DESCRIPTION
**Pull request by zuzannaszymanda**

### CODE REVIEW

File: src/atoms/list/List.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
+.a-list {
+  @apply m-0 p-0;
+  @apply list-none;
+  @apply text-primary;
```
Comment (szafran89) [inline src/atoms/list/List.css]: `@apply list-none;` and  `@apply text-primary;` are from typography section. Should it be in the same line?

File: src/atoms/list/List.css [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+.a-list--horizontal {
+  @apply flex;
+  @apply flex-row flex-wrap;
+}
+
+@screen md {
+  .a-list--horizontal {
+    @apply flex-no-wrap;
+  }
+}
```
Comment (szafran89) [inline src/atoms/list/List.css]: Are you sure we need this styles for screen md?

File: src/atoms/list/List.stories.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+export default {
+  title: 'Atoms/List',
+  component: AList
+}
+
+export const Default = () => ({
+  components: { AList, AListItem },
+  props: {
+    classKnobs: {
+      default: select('BEM Modifier', classKnobsConfig)
```
Comment (szafran89) [inline src/atoms/list/List.stories.js]: can you use multiselect instead of select to set multiple classes

File: src/atoms/list/List.css [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  @apply text-center;
+}
+
+.a-list--native {
+  @apply list-disc;
+}
+
+.a-list--native.a-list--horizontal {
+  @apply list-none;
+}
```
Comment (szafran89) [inline src/atoms/list/List.css]: IMO we don't need this. If someone use `horizontal` he should not use native. If I'm wrong let me know :)

File: src/atoms/list/List.stories.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          Lorem ipsum
+        </a-list-item>
+        <a-list-item>
+          Lorem ipsum
+        </a-list-item>
+      </a-list>
+    `
+})
+
+export const Description = () => ({
```
Comment (szafran89) [inline src/atoms/list/List.stories.js]: Can we remove this space before colon?

**Review by szafran89 — CHANGES_REQUESTED** [review]

File: src/atoms/list/List.css (hunk 4 above)
Comment (zuzannaszymanda) [reply src/atoms/list/List.css]: it's more for safety reasons than to actually use it, I can remove it

File: utils/helpers/get-select-knobs-config.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export default (items) => {
+  items = items.map((item) => item.name.replace('.', ''))
+  items.unshift(null)
```
Comment (szafran89) [inline utils/helpers/get-select-knobs-config.js]: we don't need null value in select

there is issue in console
image

File: src/atoms/list/List.stories.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+export default {
+  title: 'Atoms/List',
+  component: AList
+}
+
+export const Default = () => ({
+  components: { AList, AListItem },
+  props: {
+    classKnobs: {
+      default: optionsKnob('BEM Modifier', classKnobsConfig, 'null', { display: 'multi-select' })
```
Comment (szafran89) [inline src/atoms/list/List.stories.js]: it's not good idea to add null as string as default value.

File: src/atoms/list/List.stories.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    `
+})
+
+export const Description = () => ({
+  components: { AList, AListItem },
+  props: {
+    tagKnobs: {
+      default: text('Tag', 'dl')
+    }
+  },
```
Comment (szafran89) [inline src/atoms/list/List.stories.js]: please remove this knob. This tag is required

File: src/atoms/list/List.stories.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  components: { AList, AListItem },
+  props: {
+    classKnobs: {
+      default: optionsKnob('BEM Modifier', classKnobsConfig, 'null', { display: 'multi-select' })
+    }
+  },
+  template: `
+    <a-list :class="classKnobs">
+    </a-list>
+  `
```
Comment (szafran89) [inline src/atoms/list/List.stories.js:73]: please add example content

**Comment by szafran89** [comment]
- can't see preview for `with slot` story 
- we should not change tags for description list dl > dt is required http://prntscr.com/ufg7ux
- we have to remove this space http://prntscr.com/ufg8fc
- please add all modifiers as new story. It will be more readable IMO

File: src/atoms/list-item/ListItem.css [hunk 10]
```diff
@@ -0,0 +1,20 @@
+.a-list-item {
+  @apply p-2;
+}
+
+.a-list-item--label {
+  float: left;
```
Comment (szafran89) [inline src/atoms/list-item/ListItem.css]: if you need this value you can remove from config `float: false`

**Review by szafran89 — CHANGES_REQUESTED** [review]

File: src/atoms/list/List.stories.js [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+//         Lorem ipsum
+//       </a-list-item>
+//       <a-list-item tag="dd">
+//         Lorem ipsum value
+//       </a-list-item>
+//     </a-list>
+//   `
+// })
+
+// export const Description = DescriptionTemplate.bind({})
```
Comment (ofrankowska) [inline src/atoms/list/List.stories.js]: ? :)

File: utils/helpers/alpaca-ui.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
+var alpacaUIMixin = {
+  computed: {
+    alpacaConfig () {
+      const config = this.$options.config
+      return config
```
Comment (ofrankowska) [inline utils/helpers/alpaca-ui.js:6]: can you add an empty line before return?

File: utils/helpers/alpaca-ui.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  }
+}
+exports.alpacaUIMixin = alpacaUIMixin
+
+function AlpacaUI (Vue) {
+  Vue.mixin(alpacaUIMixin)
+}
+
+var _default = AlpacaUI
+exports.default = _default
```
Comment (ofrankowska) [inline utils/helpers/alpaca-ui.js]: why do we need two different exports?

File: utils/helpers/get-select-knobs-config.js [hunk 14]
```diff
@@ -0,0 +1,7 @@
+export default (items) => {
+  items = items.map((item) => item.name.replace('.', ''))
+  return items.reduce((obj, item) => {
+    obj[item] = item
+    return obj
+  }, {})
+}
```
Comment (ofrankowska) [inline utils/helpers/get-select-knobs-config.js]: where is this used?

**Review by ofrankowska — CHANGES_REQUESTED** [review]

File: utils/helpers/alpaca-ui.js (hunk 12 above)
Comment (zuzannaszymanda) [reply utils/helpers/alpaca-ui.js:6]: sure

File: src/atoms/list/List.stories.js (hunk 11 above)
Comment (zuzannaszymanda) [reply src/atoms/list/List.stories.js]: removed ;)

File: utils/helpers/get-select-knobs-config.js (hunk 14 above)
Comment (zuzannaszymanda) [reply utils/helpers/get-select-knobs-config.js]: Nowhere I guess

File: utils/helpers/alpaca-ui.js (hunk 13 above)
Comment (zuzannaszymanda) [reply utils/helpers/alpaca-ui.js]: I changed it to default

**Review by ofrankowska — APPROVED** [review]

**Pull request merged**